subsys/net/ip/Kconfig.ipv4

```diff
@@ -34,6 +34,11 @@ config NET_DHCPV4
 	depends on NET_IPV4
 	default n
 
+config NET_IPV4_AUTOCONF
```

Comment:
As you are introducing a new kconfig option for net, could you also add that option to tests/net/all/prj.conf file which we are using as a rudimentary way to confirm that there are no compile errors if that option is enabled.